src/auth/roles/roles.guard.ts

```diff
@@ -1,18 +1,22 @@
-import { CanActivate, ExecutionContext, Injectable } from "@nestjs/common";
-import { Reflector } from "@nestjs/core";
-import { Observable } from "rxjs";
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common'
+import { Reflector } from '@nestjs/core'
+import { Observable } from 'rxjs'
 
 @Injectable()
 export class RolesGuard implements CanActivate {
     constructor(private reflector: Reflector) {}
-    
-    canActivate(context: ExecutionContext): boolean | Promise<boolean> | Observable<boolean> {
-        const requiredRole = this.reflector.getAllAndOverride<string[]>('role', [context.getHandler(), context.getClass()])
 
-        if (!requiredRole) 
-            return true
+    canActivate(
+        context: ExecutionContext,
+    ): boolean | Promise<boolean> | Observable<boolean> {
+        const requiredRole = this.reflector.getAllAndOverride<string[]>(
+            'role',
+            [context.getHandler(), context.getClass()],
+        )
+
+        if (!requiredRole) return true
 
         const { user } = context.switchToHttp().getRequest()
```

Comment:
If we do a lockup for the user to validate the JWT we could also check the role directly. With that we can ensure that users always have the correct role applied. 

Example:

Devote admin will not affect his token. He has to relogin.